### PR TITLE
Replace Spring Cloud libraries

### PR DESCRIPTION
**Pull request by nictas**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
         <spring-security.version>5.3.4.RELEASE</spring-security.version>
         <spring-security-oauth2.version>2.5.0.RELEASE</spring-security-oauth2.version>
         <spring-security-jwt.version>1.1.1.RELEASE</spring-security-jwt.version>
-        <spring-cloud.version>2.0.7.RELEASE</spring-cloud.version>
+        <hikari.version>2.4.13</hikari.version>
```
Comment (theghost5800) [inline pom.xml:40]: There exists newer versions for java 8 and 11 https://github.com/brettwooldridge/HikariCP

File: multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/util/VcapServiceFinder.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        }
+        try {
+            return env.findJdbcServiceByName(name);
+        } catch (IllegalArgumentException e) { // Thrown when there are 0 or more matches for the specified name.
+            return null;
+        }
+    }
+
+    public CfService findService(String name) {
+        if (StringUtils.isEmpty(name)) {
```
Comment (theghost5800) [inline multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/util/VcapServiceFinder.java:29]: Same comment

File: multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/bean/factory/FileSystemFileStorageFactoryBean.java [hunk 3]
```diff
@@ -1,69 +1,60 @@
...
+        return getStoragePath(service);
+    }
+
+    @SuppressWarnings("unchecked")
+    private String getStoragePath(CfService service) {
+        Map<String, Object> credentials = service.getCredentials()
+                                                 .getMap();
+        List<Object> volumeMounts = (List<Object>) credentials.get("volume_mounts");
+        Map<String, Object> volumeMount = (Map<String, Object>) volumeMounts.get(0);
+        return (String) volumeMount.get("container_dir");
```
[the remark this answers is not in the document]
Comment (nictas) [reply multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/bean/factory/FileSystemFileStorageFactoryBean.java:57]: I don't know. This is unchanged code from: https://github.com/cloudfoundry-incubator/multiapps-controller/blob/4f9ad9d6dd0442ed28c7a75e9c81bb74916e9626/multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/service/FileSystemServiceInfoCreator.java#L31

File: multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/util/VcapServiceFinder.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import io.pivotal.cfenv.jdbc.CfJdbcEnv;
+import io.pivotal.cfenv.jdbc.CfJdbcService;
+
+@Named
+public class VcapServiceFinder {
+
+    private final CfJdbcEnv env = new CfJdbcEnv();
+
+    public CfJdbcService findJdbcService(String name) {
+        if (StringUtils.isEmpty(name)) {
```
[the remark this answers is not in the document]
Comment (nictas) [reply multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/util/VcapServiceFinder.java:18]: No idea. Same code as in: https://github.com/cloudfoundry-incubator/multiapps-controller/blob/4f9ad9d6dd0442ed28c7a75e9c81bb74916e9626/multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/bean/factory/CloudDataSourceFactoryBean.java#L59
My goal was to change as little as possible.

File: multiapps-controller-web/pom.xml [hunk 5]
```diff
@@ -203,5 +194,17 @@
...
+            <groupId>io.pivotal.cfenv</groupId>
+            <artifactId>java-cfenv</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.pivotal.cfenv</groupId>
+            <artifactId>java-cfenv-jdbc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.zaxxer</groupId>
+            <artifactId>HikariCP-java7</artifactId>
```
[the remark this answers is not in the document]
Comment (nictas) [reply multiapps-controller-web/pom.xml:207]: Quartz uses HikariCP-java7, and we cannot upgrade to a newer version as a result.

File: multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/service/ObjectStoreServiceInfo.java [hunk 6]
```diff
@@ -1,103 +1,23 @@
 package org.cloudfoundry.multiapps.controller.web.configuration.service;
 
-import org.springframework.cloud.service.BaseServiceInfo;
-import org.springframework.cloud.service.ServiceInfo.ServiceLabel;
+import org.cloudfoundry.multiapps.common.Nullable;
+import org.immutables.value.Value;
 
-@ServiceLabel("objectstore")
-public class ObjectStoreServiceInfo extends BaseServiceInfo {
+@Value.Immutable
```
Comment (radito3) [inline multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/service/ObjectStoreServiceInfo.java:6]: Nice

**Review by theghost5800 — APPROVED** [review]

**Review by radito3 — APPROVED** [review]

**Pull request merged**